File: command-pattern/java/src/CommandInvoker.java
package src;

import java.util.*;

// 命令调用类，通过关联命令来实行命令的调用
public class CommandInvoker {

    private List<Command> commandList = new ArrayList<Command>();

    // 储存命令
    public void takeOrder(Command command) {
        System.out.println(this.getClass().getName() + "::takeOrder() " + command.getClass().getName());
        commandList.add(command);
    }

    // 统一执行
    public void executeOrders() {
        System.out.println(this.getClass().getName() + "::executeOrders() ");
        for (Command command : commandList) {
            command.execute();
        }
        commandList.clear();
    }
}

File: interpreter-pattern/python/src/CommandInvoker.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

# 命令调用类，通过关联命令来实行命令的调用
class CommandInvoker:
    def __init__(self):
        self.commandList = []

    # 储存命令
    def take_order(self, command):
        print(
            self.__class__.__name__ + '::takeOrder() ' + command.__class__.__name__
        )
        self.commandList.append(command)

    # 统一执行
    def execute_orders(self):
        print(self.__class__.__name__ + '::executeOrders() ')
        for command in self.commandList:
            command.execute()

        self.commandList = []
